### ISSUE TITLE
Scouts ignore unit overlaping

### ISSUE THREAD

**Issue by Piotr-Wieczorek** [LoneGazebo/Community-Patch-DLL#11456]
_1. Mod Version (X.Y.Z). Current Version: 4.17.8_

4.17.8

_2. DirectX Version_

DX11 + DXVK

_3. List of Other Mods_

https://steamcommunity.com/sharedfiles/filedetails/?id=220250184, https://steamcommunity.com/sharedfiles/filedetails/?id=166166333, https://steamcommunity.com/sharedfiles/filedetails/?id=83056827

_4. Describe the Issue_

Scouts stop on other units and doesn't need to be moved (even when the game says that it has to be moved).

______________________________________________________________________________________
_5. Save Game From 1 Turn Before (ALWAYS ATTACH THIS IF POSSIBLE)_

/home/piotr/.steam/debian-installation/steamapps/compatdata/8930/pfx/drive_c/users/steamuser/Documents/My Games/Sid Meier's Civilization 5/ModdedSaves/single/auto/AutoSave_0244 AD-0990.Civ5Save.zip

_6. Logs (ALWAYS ATTACH THESE IF POSSIBLE)_

Impossible, they weigh 800 MB.

_7. CvMiniDump.dmp File (ATTACH IF REPORTING A GAME CRASH)_

Not a crash

_8. Steps to reproduce the Issue (Optional)_

Try to move the scout to the leftmost tile behind the composite bowman.

_9. Screenshots of the Issue (Optional)_
![obraz](https://github.com/user-attachments/assets/5acc2dd1-d4a6-40ea-af05-e955b5a5270e)

**Comment by azum4roll** [comment]
There's no save attached.

**Comment by Piotr-Wieczorek** [comment]
I have accidentally pasted the path of the save instead of the save itself. 
[AutoSave_0244 AD-0990.Civ5Save.zip](https://github.com/user-attachments/files/18221877/AutoSave_0244.AD-0990.Civ5Save.zip)

**Comment by bsesar** [comment]
@Piotr-Wieczorek, have you tried to end the turn with the scout overlapping another unit's position?

**Comment by Piotr-Wieczorek** [comment]
@bsesar Yes, the game allows both units to stay and heal (archer must fortify to heal before the overlap occurs). It seems that this can occur only with ranged units, my scout has never stopped and overlapped a melee unit.
